pkgs/by-name/wm/wml/package.nix

```diff
@@ -1,92 +1,81 @@
 {
+  cmake,
+  fetchFromGitHub,
   lib,
-  fetchurl,
-  perlPackages,
-  ncurses,
   lynx,
-  makeWrapper,
+  makeBinaryWrapper,
+  ncurses,
+  pcre,
+  perl,
+  perlPackages,
+  stdenv,
 }:
-
-perlPackages.buildPerlPackage {
+let
+  shlomif-cmake-modules = fetchFromGitHub {
+    owner = "shlomif";
+    repo = "shlomif-cmake-modules";
+    rev = "2fa3e9be1a1df74ad0e10f0264bfa60e1e3a755c";
+    hash = "sha256-MNGpegbZRwfD8A3VHVNYrDULauLST3Nt18/3Ht6mpZw=";
+  };
+  perlDeps = with perlPackages; [
+    BitVector
+    CarpAlways
+    ClassXSAccessor
+    FileWhich
+    GD
+    ImageSize
+    ListMoreUtils
+    PathTiny
+    TermReadKey
+  ];
+in
+stdenv.mkDerivation (finalAttrs: {
   pname = "wml";
-  version = "2.0.11";
+  version = "2.32.0";
 
-  src = fetchurl {
-    url = "http://thewml.org/distrib/wml-2.0.11.tar.gz";
-    sha256 = "0jjxpq91x7y2mgixz7ghqp01m24qa37wl3zz515rrzv7x8cyy4cf";
+  src = fetchFromGitHub {
+    owner = "thewml";
+    repo = "website-meta-language";
+    tag = "releases/wml-${finalAttrs.version}";
+    hash = "sha256-9ZiMGm0W2qS/7nL8NsmGBsuB5sNJvWuJaxE7CTdWo6s=";
   };
 
-  setOutputFlags = false;
-
-  # Getting lots of Non-ASCII character errors from pod2man.
-  # Inserting =encoding utf8 before the first =head occurrence.
-  # Wasn't able to fix mp4h.
-  preConfigure = ''
-    touch Makefile.PL
-    for i in wml_backend/p6_asubst/asubst.src wml_aux/freetable/freetable.src wml_docs/*.pod wml_include/des/*.src wml_include/fmt/*.src; do
-      sed -i '0,/^=head/{s/^=head/=encoding utf8\n=head/}' $i
-    done
-    sed -i 's/ doc / /g' wml_backend/p2_mp4h/Makefile.in
-    sed -i '/p2_mp4h\/doc/d' Makefile.in
-  '';
+  sourceRoot = "${finalAttrs.src.name}/src";
 
-  nativeBuildInputs = [ makeWrapper ];
-  buildInputs = with perlPackages; [
-    perl
-    TermReadKey
-    GD
-    BitVector
-    ncurses
+  nativeBuildInputs = [
+    cmake
     lynx
-    ImageSize
+    makeBinaryWrapper
   ];
 
-  patches = [
-    ./redhat-with-thr.patch
-    ./dynaloader.patch
-    ./no_bitvector.patch
+  buildInputs = perlDeps ++ [
+    ncurses
+    pcre
+    perl
   ];
 
-  # Workaround build failure on -fno-common toolchains:
-  #   ld: iselect_browse.o:(.bss+0x2020): multiple definition of `Line'; iselect_main.o:(.bss+0x100000): first defined here
-  env.NIX_CFLAGS_COMPILE = "-fcommon";
+  preConfigure = ''
+    ln -s ${shlomif-cmake-modules}/shlomif-cmake-modules/Shlomif_Common.cmake cmake/
+  '';
 
-  hardeningDisable = [ "format" ];
+  cmakeFlags = [ (lib.cmakeFeature "CMAKE_POLICY_VERSION_MINIMUM" "3.5") ];
```

Comment:
please prefer a patch, or at least `substituteInPlace --replace-fail`. `CMAKE_POLICY_VERSION_MINIMUM` as a flag runs the risk of later *downgrading* cmake: Upstream might upgrade their cmake version requirements, but the flag will override that back down to 3.5. A patch or substitute will instead loudly complain once the substitution fails, leading to actual useful build-time errors during the update and less pain of forgotten code.

Also prefer bumping straight to 3.10, as 3.5 is already deprecated by cmake upstream.